### PR TITLE
feat: filter votes by choice

### LINKED ISSUE #290: [GraphQL]  Filter votes by choice
Currently there is no way to filter votes by choice. There should be a where-filter taking an array of choices to return.

### PR DESCRIPTION
**Pull request by mktcode**
See https://github.com/snapshot-labs/snapshot-hub/issues/290#issuecomment-1113165105

but this breaks pagination :(

### CODE REVIEW

File: src/graphql/operations/votes.ts [hunk 1]
```diff
@@ -56,6 +56,39 @@ async function query(parent, args, context?, info?) {
...
+      }
+      if (typeof v.choice === 'object') {
+        if (Array.isArray(v.choice)) {
+          return v.choice.includes(where.choice);
+        } else {
+          return Object.keys(v.choice).includes(where.choice.toString());
+        }
+      }
+      return false;
+    });
```
Comment (ChaituVR) [inline src/graphql/operations/votes.ts:73]: This may affect `first` and `skip` for pagination right?

Comment (ChaituVR) [reply src/graphql/operations/votes.ts:73]: Maybe we should have a new input variable `filter` instead of `where`

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/graphql/operations/votes.ts (hunk 1 above)
Comment (mktcode) [reply src/graphql/operations/votes.ts:73]: You mean as a clear way of saying "this filtering happens after pagination"?

For a moment I thought about doing the pagination in JS too. Removing the `LIMIT` from the query, at least when there's a choice filter. But that doesn't feel right.

So yea... idk. We just tell people it's not possible and they have to do the filtering themselves, in their client?

Comment (bonustrack) [reply src/graphql/operations/votes.ts:73]: @mktcode Yes, they can filter on their side, we would implement this when we have an elegant way to do it. Who is asking this feature?

**Comment by bonustrack** [comment]
Closing this PR for now, it's unlikely we will allow query for specific choice as the choice field is a JSON and query into doesn't scale well.

**Pull request closed without merging**